diff --git a/libs/hwui/jni/android_graphics_HardwareRendererObserver.cpp b/libs/hwui/jni/android_graphics_HardwareRendererObserver.cpp
index 6cae5ffa..b98a46cd 100644
--- a/libs/hwui/jni/android_graphics_HardwareRendererObserver.cpp
+++ b/libs/hwui/jni/android_graphics_HardwareRendererObserver.cpp
@@ -114,7 +114,7 @@ static jint android_graphics_HardwareRendererObserver_getNextBuffer(JNIEnv* env,
     }
 }
 
-static const std::array gMethods = {
+static const JNINativeMethod gMethods[] = {
         MAKE_JNI_NATIVE_METHOD("nCreateObserver", "(Ljava/lang/ref/WeakReference;Z)J",
                                android_graphics_HardwareRendererObserver_createObserver),
         MAKE_JNI_NATIVE_METHOD("nGetNextBuffer", "(J[J)I",
@@ -130,7 +130,7 @@ int register_android_graphics_HardwareRendererObserver(JNIEnv* env) {
             env, observerClass, "invokeDataAvailable", "(Ljava/lang/ref/WeakReference;)Z");
 
     return RegisterMethodsOrDie(env, "android/graphics/HardwareRendererObserver",
-                                gMethods.data(), gMethods.size());
+                                gMethods, NELEM(gMethods));
 
 }
 
