diff --git a/libs/hwui/jni/android_graphics_TextureLayer.cpp b/libs/hwui/jni/android_graphics_TextureLayer.cpp
index 4dbb24ce..27618afb 100644
--- a/libs/hwui/jni/android_graphics_TextureLayer.cpp
+++ b/libs/hwui/jni/android_graphics_TextureLayer.cpp
@@ -1,3 +1,4 @@
+#if 0  // OH adapter TextureLayer stub
 /*
  * Copyright (C) 2014 The Android Open Source Project
  *
@@ -83,3 +84,5 @@ int register_android_graphics_TextureLayer(JNIEnv* env) {
 }
 
 };
+
+#endif
