diff --git a/libs/hwui/jni/BitmapRegionDecoder.cpp b/libs/hwui/jni/BitmapRegionDecoder.cpp
index 740988f7..7f0e65f9 100644
--- a/libs/hwui/jni/BitmapRegionDecoder.cpp
+++ b/libs/hwui/jni/BitmapRegionDecoder.cpp
@@ -1,3 +1,4 @@
+#if 0  // OH adapter BRD stub
 /*
  * Copyright (C) 2010 The Android Open Source Project
  *
@@ -430,3 +431,5 @@ int register_android_graphics_BitmapRegionDecoder(JNIEnv* env)
     return android::RegisterMethodsOrDie(env, "android/graphics/BitmapRegionDecoder",
             gBitmapRegionDecoderMethods, NELEM(gBitmapRegionDecoderMethods));
 }
+
+#endif
