diff --git a/libs/hwui/jni/GIFMovie.cpp b/libs/hwui/jni/GIFMovie.cpp
index ae6ac4ce..faf6458b 100644
--- a/libs/hwui/jni/GIFMovie.cpp
+++ b/libs/hwui/jni/GIFMovie.cpp
@@ -1,3 +1,4 @@
+#if 0  // OH adapter: OH adapter stub GIFMovie.cpp
 /*
  * Copyright 2006 The Android Open Source Project
  *
@@ -450,3 +451,5 @@ Movie* Movie::DecodeStream(SkStreamRewindable* stream) {
     }
     return nullptr;
 }
+
+#endif
