--- a/libs/hwui/jni/graphics_jni_helpers.h
+++ b/libs/hwui/jni/graphics_jni_helpers.h
@@ -27,7 +27,7 @@
 // and they need all the JNI methods to have JNIEnv* and jclass/jobject as their first two arguments.
 // The following macro allows to have those arguments when compiling for host while omitting them when
 // compiling for Android.
-#ifdef __ANDROID__
+#if defined(__ANDROID__) || defined(__OHOS__)
 #define CRITICAL_JNI_PARAMS
 #define CRITICAL_JNI_PARAMS_COMMA
 #else
