diff --git a/libs/hwui/jni/GraphicsStatsService.cpp b/libs/hwui/jni/GraphicsStatsService.cpp
index e32c9115..f2c2fbd0 100644
--- a/libs/hwui/jni/GraphicsStatsService.cpp
+++ b/libs/hwui/jni/GraphicsStatsService.cpp
@@ -136,7 +136,7 @@ static AStatsManager_PullAtomCallbackReturn graphicsStatsPullCallback(int32_t at
     }
     if (gGraphicsStatsServiceObject == nullptr) {
         ALOGE("Failed to get graphicsstats service");
-        return AStatsManager_PULL_SKIP;
+        return 1;
     }
 
     for (bool lastFullDay : {true, false}) {
@@ -148,7 +148,7 @@ static AStatsManager_PullAtomCallbackReturn graphicsStatsPullCallback(int32_t at
             env->ExceptionDescribe();
             env->ExceptionClear();
             ALOGE("Failed to invoke graphicsstats service");
-            return AStatsManager_PULL_SKIP;
+            return 1;
         }
     }
     return AStatsManager_PULL_SUCCESS;
@@ -157,18 +157,17 @@ static AStatsManager_PullAtomCallbackReturn graphicsStatsPullCallback(int32_t at
 // Register a puller for GRAPHICS_STATS atom with the statsd service.
 static void nativeInit(JNIEnv* env, jobject javaObject) {
     gGraphicsStatsServiceObject = env->NewGlobalRef(javaObject);
-    AStatsManager_PullAtomMetadata* metadata = AStatsManager_PullAtomMetadata_obtain();
-    AStatsManager_PullAtomMetadata_setCoolDownMillis(metadata, 10);             // 10 milliseconds
-    AStatsManager_PullAtomMetadata_setTimeoutMillis(metadata, 2 * MS_PER_SEC);  // 2 seconds
+    AStatsManager_PullAtomMetadata* metadata = nullptr;
+    /* removed */;             // 10 milliseconds
+    /* removed */;  // 2 seconds
 
-    AStatsManager_setPullAtomCallback(stats::GRAPHICS_STATS, metadata, &graphicsStatsPullCallback,
-                                      nullptr);
+    /* removed */
 
-    AStatsManager_PullAtomMetadata_release(metadata);
+    /* removed */;
 }
 
 static void nativeDestructor(JNIEnv* env, jobject javaObject) {
-    AStatsManager_clearPullAtomCallback(stats::GRAPHICS_STATS);
+    /* removed */;
     env->DeleteGlobalRef(gGraphicsStatsServiceObject);
     gGraphicsStatsServiceObject = nullptr;
 }
