diff --git a/libs/hwui/jni/GraphicsJNI.h b/libs/hwui/jni/GraphicsJNI.h
index 23ab5dd3..fb938e4d 100644
--- a/libs/hwui/jni/GraphicsJNI.h
+++ b/libs/hwui/jni/GraphicsJNI.h
@@ -188,7 +188,7 @@ public:
         return mStorage.release();
     };
 
-    SkCodec::ZeroInitialized zeroInit() const override { return SkCodec::kYes_ZeroInitialized; }
+    SkCodec::ZeroInitialized zeroInit() const /* override removed */ { return SkCodec::kYes_ZeroInitialized; }
 private:
     sk_sp<android::Bitmap> mStorage;
 };
@@ -244,7 +244,7 @@ public:
      *  Indicates that this allocator does not allocate zero initialized
      *  memory.
      */
-    SkCodec::ZeroInitialized zeroInit() const override { return SkCodec::kNo_ZeroInitialized; }
+    SkCodec::ZeroInitialized zeroInit() const /* override removed */ { return SkCodec::kNo_ZeroInitialized; }
 
 private:
     android::Bitmap* mRecycledBitmap;
