diff --git a/libs/hwui/jni/MaskFilter.cpp b/libs/hwui/jni/MaskFilter.cpp
index 048ce025..fa8fcacb 100644
--- a/libs/hwui/jni/MaskFilter.cpp
+++ b/libs/hwui/jni/MaskFilter.cpp
@@ -35,8 +35,7 @@ public:
         }
 
         SkScalar sigma = SkBlurMask::ConvertRadiusToSigma(radius);
-        SkMaskFilter* filter =  SkBlurMaskFilter::MakeEmboss(sigma,
-                direction, ambient, specular).release();
+        SkMaskFilter* filter =  nullptr; /* M133: MakeEmboss removed */
         ThrowIAE_IfNull(env, filter);
         return reinterpret_cast<jlong>(filter);
     }
