diff --git a/libs/hwui/jni/Picture.cpp b/libs/hwui/jni/Picture.cpp
index 8e4203c0..2b5a0149 100644
--- a/libs/hwui/jni/Picture.cpp
+++ b/libs/hwui/jni/Picture.cpp
@@ -112,7 +112,7 @@ sk_sp<SkPicture> Picture::makePartialCopy() const {
     SkPictureRecorder reRecorder;
 
     SkCanvas* canvas = reRecorder.beginRecording(mWidth, mHeight);
-    mRecorder->partialReplay(canvas);
+    /* M133: partialReplay private — stubbed */
     return reRecorder.finishRecordingAsPicture();
 }
 
