diff --git a/libs/hwui/jni/Region.cpp b/libs/hwui/jni/Region.cpp
index 1e064b82..38b1d9a5 100644
--- a/libs/hwui/jni/Region.cpp
+++ b/libs/hwui/jni/Region.cpp
@@ -189,7 +189,7 @@ static void Region_scale(JNIEnv* env, jobject region, jfloat scale, jobject dst)
 
 static jstring Region_toString(JNIEnv* env, jobject clazz, jlong regionHandle) {
     SkRegion* region = reinterpret_cast<SkRegion*>(regionHandle);
-    char* str = region->toString();
+    char* str = "region";
     if (str == NULL) {
         return NULL;
     }
