diff --git a/libs/hwui/pipeline/skia/StretchMask.cpp b/libs/hwui/pipeline/skia/StretchMask.cpp
index cad3703d..ea292adb 100644
--- a/libs/hwui/pipeline/skia/StretchMask.cpp
+++ b/libs/hwui/pipeline/skia/StretchMask.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 #include "StretchMask.h"
+#include <gpu/ganesh/SkSurfaceGanesh.h>
 
 #include "SkBlendMode.h"
 #include "SkCanvas.h"
@@ -36,7 +37,7 @@ void StretchMask::draw(GrRecordingContext* context,
         mMaskSurface->height() != height) {
         // Create a new surface if we don't have one or our existing size does
         // not match.
-        mMaskSurface = SkSurface::MakeRenderTarget(
+        mMaskSurface = SkSurfaces::RenderTarget(
             context,
             skgpu::Budgeted::kYes,
             SkImageInfo::Make(
