diff --git a/libs/hwui/CanvasTransform.cpp b/libs/hwui/CanvasTransform.cpp
index cd4fae86..fd7a7ed2 100644
--- a/libs/hwui/CanvasTransform.cpp
+++ b/libs/hwui/CanvasTransform.cpp
@@ -125,7 +125,7 @@ static BitmapPalette filterPalette(const SkPaint* paint, BitmapPalette palette)
     }
 
     SkColor color = palette == BitmapPalette::Light ? SK_ColorWHITE : SK_ColorBLACK;
-    color = paint->getColorFilter()->filterColor(color);
+    color = color /* OH adapter: filterColor stub */;
     return paletteForColorHSV(color);
 }
 
