diff --git a/libs/hwui/RenderNode.cpp b/libs/hwui/RenderNode.cpp
index 1c39db3a..1c370eed 100644
--- a/libs/hwui/RenderNode.cpp
+++ b/libs/hwui/RenderNode.cpp
@@ -15,6 +15,7 @@
  */
 
 #include "RenderNode.h"
+#include <gpu/ganesh/SkImageGanesh.h>
 
 #include "DamageAccumulator.h"
 #include "Debug.h"
@@ -358,7 +359,7 @@ std::optional<RenderNode::SnapshotResult> RenderNode::updateSnapshotIfRequired(
                mTargetImageFilterLayerSurfaceGenerationId != layerSurfaceGenerationId) {
         // Otherwise create a new snapshot with the given filter and snapshot
         mSnapshotResult.snapshot =
-                snapshot->makeWithFilter(context,
+                SkImages::MakeWithFilter(context, snapshot,
                                          imageFilter,
                                          subset,
                                          clipBounds,
