diff --git a/libs/hwui/WebViewFunctorManager.cpp b/libs/hwui/WebViewFunctorManager.cpp
index 6fc251dc..a8d83c9b 100644
--- a/libs/hwui/WebViewFunctorManager.cpp
+++ b/libs/hwui/WebViewFunctorManager.cpp
@@ -228,7 +228,7 @@ ASurfaceControl* WebViewFunctor::getSurfaceControl() {
     activeContext->prepareSurfaceControlForWebview();
     funcs.transactionSetZOrderFunc(transaction, mSurfaceControl, -1);
     funcs.transactionSetVisibilityFunc(transaction, mSurfaceControl,
-                                       ASURFACE_TRANSACTION_VISIBILITY_SHOW);
+                                       1);
     funcs.transactionApplyFunc(transaction);
     funcs.transactionDeleteFunc(transaction);
     return mSurfaceControl;
