--- a/base/startup/appspawn/modules/module_engine/include/appspawn_msg.h
+++ b/base/startup/appspawn/modules/module_engine/include/appspawn_msg.h
@@ -34,5 +34,6 @@
 #define SPAWN_LISTEN_FD_NAME "SpawnListenFd"
 #define HYBRIDSPAWN_SOCKET_NAME "HybridSpawn"
+#define APPSPAWNX_SOCKET_NAME "AppSpawnX"  // adapter project: Android APK spawner
 
 #define APPSPAWN_ALIGN(len) (((len) + 0x03) & (~0x03))
 #define APPSPAWN_TLV_NAME_LEN 32
