--- a/libartbase/base/metrics/metrics.h
+++ b/libartbase/base/metrics/metrics.h
@@ -336,7 +336,7 @@
   bool IsNull() const override { return Value() == 0; }
 
   std::atomic<value_t> value_;
-  static_assert(std::atomic<value_t>::is_always_lock_free);
+  // static_assert(std::atomic<value_t>::is_always_lock_free); // ARM32 compat
 
   friend class ArtMetrics;
 };
@@ -390,7 +390,7 @@
   bool IsNull() const override { return Count() == 0; }
 
   std::atomic<count_t> count_;
-  static_assert(std::atomic<count_t>::is_always_lock_free);
+  // static_assert(std::atomic<count_t>::is_always_lock_free); // ARM32 compat
 
   friend class ArtMetrics;
 };
@@ -428,7 +428,7 @@
   bool IsNull() const override { return Value() == 0; }
 
   std::atomic<value_t> value_;
-  static_assert(std::atomic<value_t>::is_always_lock_free);
+  // static_assert(std::atomic<value_t>::is_always_lock_free); // ARM32 compat
 
   friend class ArtMetrics;
 };
@@ -498,7 +498,7 @@
   }
 
   std::array<std::atomic<value_t>, num_buckets_> buckets_;
-  static_assert(std::atomic<value_t>::is_always_lock_free);
+  // static_assert(std::atomic<value_t>::is_always_lock_free); // ARM32 compat
 
   friend class ArtMetrics;
 };
