--- a/runtime/gc/heap-inl.h
+++ b/runtime/gc/heap-inl.h
@@ -304,7 +304,7 @@
                                            size_t* usable_size,
                                            size_t* bytes_tl_bulk_allocated) {
   if (allocator_type != kAllocatorTypeRegionTLAB &&
-      allocator_type != kAllocatorTypeTLAB &&
+        allocator_type != kAllocatorTypeTLAB &&
       allocator_type != kAllocatorTypeRosAlloc &&
       UNLIKELY(IsOutOfMemoryOnAllocation(allocator_type, alloc_size, kGrow))) {
     return nullptr;
