--- a/runtime/thread.h
+++ b/runtime/thread.h
@@ -1893,7 +1893,7 @@
     // A count of the number of readers of tlsPtr_.name that may still be looking at a string they
     // retrieved.
     mutable std::atomic<uint32_t> num_name_readers;
-    static_assert(std::atomic<uint32_t>::is_always_lock_free);
+    // static_assert(std::atomic<uint32_t>::is_always_lock_free); // ARM32 compat
 
     // Thread-local hotness counter for shared memory methods. Initialized with
     // `kSharedMethodHotnessThreshold`. The interpreter decrements it and goes
@@ -2044,7 +2044,7 @@
     // If this is not null or kThreadNameDuringStartup, then it owns the malloc memory holding
     // the string. Updated in an RCU-like manner.
     std::atomic<const char*> name;
-    static_assert(std::atomic<const char*>::is_always_lock_free);
+    // static_assert(std::atomic<const char*>::is_always_lock_free); // ARM32 compat
 
     // A cached pthread_t for the pthread underlying this Thread*.
     pthread_t pthread_self;
