diff --git a/adbconnection/Android.bp b/adbconnection/Android.bp
index ecad650..f5cb234 100644
--- a/adbconnection/Android.bp
+++ b/adbconnection/Android.bp
@@ -66,6 +66,7 @@ art_cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -81,5 +82,6 @@ art_cc_library {
     ],
     apex_available: [
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
diff --git a/artd/Android.bp b/artd/Android.bp
index 1288ebc..03c52fd 100644
--- a/artd/Android.bp
+++ b/artd/Android.bp
@@ -58,6 +58,7 @@ art_cc_binary {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/build/boot/Android.bp b/build/boot/Android.bp
index 292bb6f..e86d0be 100644
--- a/build/boot/Android.bp
+++ b/build/boot/Android.bp
@@ -78,6 +78,7 @@ bootclasspath_fragment {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 
     hidden_api: {
@@ -139,5 +140,6 @@ systemserverclasspath_fragment {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
diff --git a/compiler/Android.bp b/compiler/Android.bp
index a879bd8..59fba20 100644
--- a/compiler/Android.bp
+++ b/compiler/Android.bp
@@ -249,7 +249,9 @@ art_cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
         "test_broken_com.android.art",
+        "//apex_available:platform",
     ],
 }
 
@@ -310,6 +312,7 @@ art_cc_library {
 
     apex_available: [
         "com.android.art.debug",
+        "//apex_available:platform",
         // TODO(b/183882457): This lib doesn't go into com.android.art, but
         // apex_available lists need to be the same for internal libs to avoid
         // stubs, and this depends on libdexfiled.
diff --git a/dalvikvm/Android.bp b/dalvikvm/Android.bp
index 6e06eba..ef61430 100644
--- a/dalvikvm/Android.bp
+++ b/dalvikvm/Android.bp
@@ -67,5 +67,6 @@ art_cc_binary {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
diff --git a/dex2oat/Android.bp b/dex2oat/Android.bp
index 7341774..095f2ff 100644
--- a/dex2oat/Android.bp
+++ b/dex2oat/Android.bp
@@ -116,6 +116,7 @@ art_cc_library_static {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/dexdump/Android.bp b/dexdump/Android.bp
index da43fbe..badecdf 100644
--- a/dexdump/Android.bp
+++ b/dexdump/Android.bp
@@ -70,6 +70,7 @@ art_cc_binary {
         "com.android.art",
         "com.android.art.debug",
         "test_broken_com.android.art",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/dexlayout/Android.bp b/dexlayout/Android.bp
index ec7fdf6..2c04758 100644
--- a/dexlayout/Android.bp
+++ b/dexlayout/Android.bp
@@ -281,6 +281,7 @@ art_cc_binary {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/dexlist/Android.bp b/dexlist/Android.bp
index 918fc68..186b324 100644
--- a/dexlist/Android.bp
+++ b/dexlist/Android.bp
@@ -35,6 +35,7 @@ art_cc_binary {
         "com.android.art",
         "com.android.art.debug",
         "test_broken_com.android.art",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/dexoptanalyzer/Android.bp b/dexoptanalyzer/Android.bp
index ee630a5..ceba1fc 100644
--- a/dexoptanalyzer/Android.bp
+++ b/dexoptanalyzer/Android.bp
@@ -63,6 +63,7 @@ art_cc_binary {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -78,6 +79,7 @@ art_cc_binary {
     ],
     apex_available: [
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/disassembler/Android.bp b/disassembler/Android.bp
index b7f758f..257a073 100644
--- a/disassembler/Android.bp
+++ b/disassembler/Android.bp
@@ -73,6 +73,7 @@ art_cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -107,6 +108,7 @@ art_cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/dt_fd_forward/Android.bp b/dt_fd_forward/Android.bp
index e0507d5..91e9ebf 100644
--- a/dt_fd_forward/Android.bp
+++ b/dt_fd_forward/Android.bp
@@ -81,6 +81,7 @@ art_cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/imgdiag/Android.bp b/imgdiag/Android.bp
index a0354ab..469306e 100644
--- a/imgdiag/Android.bp
+++ b/imgdiag/Android.bp
@@ -71,6 +71,7 @@ art_cc_binary {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -88,6 +89,7 @@ art_cc_binary {
     ],
     apex_available: [
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/libartbase/Android.bp b/libartbase/Android.bp
index 951599d..ff2f2f5 100644
--- a/libartbase/Android.bp
+++ b/libartbase/Android.bp
@@ -209,6 +209,7 @@ art_cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 
     target: {
@@ -426,5 +427,6 @@ cc_library_headers {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
diff --git a/libartservice/service/Android.bp b/libartservice/service/Android.bp
index 3c36b77..10ca834 100644
--- a/libartservice/service/Android.bp
+++ b/libartservice/service/Android.bp
@@ -41,6 +41,7 @@ cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
     shared_libs: [
     ],
@@ -55,6 +56,7 @@ cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
     shared_libs: [
     ],
@@ -125,6 +127,7 @@ java_sdk_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
     jarjar_rules: "jarjar-rules.txt",
     optimize: {
@@ -145,6 +148,7 @@ java_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -161,6 +165,7 @@ java_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/libarttools/Android.bp b/libarttools/Android.bp
index d38ef45..9d3d789 100644
--- a/libarttools/Android.bp
+++ b/libarttools/Android.bp
@@ -50,6 +50,7 @@ cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -115,5 +116,6 @@ cc_binary {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
diff --git a/libelffile/Android.bp b/libelffile/Android.bp
index 8281068..53ec159 100644
--- a/libelffile/Android.bp
+++ b/libelffile/Android.bp
@@ -52,6 +52,7 @@ art_cc_library_static {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/libprofile/Android.bp b/libprofile/Android.bp
index beae8a9..170b97a 100644
--- a/libprofile/Android.bp
+++ b/libprofile/Android.bp
@@ -135,6 +135,7 @@ art_cc_library {
         "com.android.art",
         "com.android.art.debug",
         "test_broken_com.android.art",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/oatdump/Android.bp b/oatdump/Android.bp
index dacd647..10b0cee 100644
--- a/oatdump/Android.bp
+++ b/oatdump/Android.bp
@@ -85,6 +85,7 @@ art_cc_binary {
         "com.android.art",
         "com.android.art.debug",
         "test_broken_com.android.art",
+        "//apex_available:platform",
     ],
 }
 
@@ -141,6 +142,7 @@ art_cc_binary {
 
     apex_available: [
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/openjdkjvm/Android.bp b/openjdkjvm/Android.bp
index eb379d8..0fe060b 100644
--- a/openjdkjvm/Android.bp
+++ b/openjdkjvm/Android.bp
@@ -66,6 +66,7 @@ art_cc_library {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -81,5 +82,6 @@ art_cc_library {
     ],
     apex_available: [
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
diff --git a/perfetto_hprof/Android.bp b/perfetto_hprof/Android.bp
index 8cfc7d4..ec2c77d 100644
--- a/perfetto_hprof/Android.bp
+++ b/perfetto_hprof/Android.bp
@@ -89,6 +89,7 @@ art_cc_library {
         "com.android.art",
         "com.android.art.debug",
         "test_broken_com.android.art",
+        "//apex_available:platform",
     ],
 }
 
@@ -105,5 +106,6 @@ art_cc_library {
     ],
     apex_available: [
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
diff --git a/profman/Android.bp b/profman/Android.bp
index 020b0de..563cbfa 100644
--- a/profman/Android.bp
+++ b/profman/Android.bp
@@ -111,6 +111,7 @@ art_cc_binary {
         "com.android.art",
         "com.android.art.debug",
         "test_broken_com.android.art",
+        "//apex_available:platform",
     ],
 }
 
@@ -138,6 +139,7 @@ art_cc_binary {
     },
     apex_available: [
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -178,6 +180,7 @@ cc_library_headers {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/runtime/Android.bp b/runtime/Android.bp
index 1c4b871..c47c8bb 100644
--- a/runtime/Android.bp
+++ b/runtime/Android.bp
@@ -106,6 +106,7 @@ cc_library_headers {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -1067,6 +1068,7 @@ cc_library_static {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/sigchainlib/Android.bp b/sigchainlib/Android.bp
index 68dec35..878dbf9 100644
--- a/sigchainlib/Android.bp
+++ b/sigchainlib/Android.bp
@@ -63,6 +63,7 @@ cc_library {
         "com.android.art",
         "com.android.art.debug",
         "test_broken_com.android.art",
+        "//apex_available:platform",
     ],
     stubs: {
         symbol_file: "libsigchain.map.txt",
diff --git a/simulator/Android.bp b/simulator/Android.bp
index d29319a..68a3113 100644
--- a/simulator/Android.bp
+++ b/simulator/Android.bp
@@ -112,5 +112,6 @@ art_cc_library {
     ],
     apex_available: [
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
diff --git a/tools/Android.bp b/tools/Android.bp
index 207a121..bb3963a 100644
--- a/tools/Android.bp
+++ b/tools/Android.bp
@@ -45,6 +45,7 @@ cc_binary {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/tools/cpp-define-generator/Android.bp b/tools/cpp-define-generator/Android.bp
index 6e8cd33..437f7d6 100644
--- a/tools/cpp-define-generator/Android.bp
+++ b/tools/cpp-define-generator/Android.bp
@@ -41,6 +41,7 @@ cc_object {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -88,6 +89,7 @@ art_module_cc_genrule {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
@@ -100,6 +102,7 @@ cc_library_headers {
     apex_available: [
         "com.android.art",
         "com.android.art.debug",
+        "//apex_available:platform",
     ],
 }
 
diff --git a/tools/dexanalyze/Android.bp b/tools/dexanalyze/Android.bp
index 5b87559..c3cec6d 100644
--- a/tools/dexanalyze/Android.bp
+++ b/tools/dexanalyze/Android.bp
@@ -50,6 +50,7 @@ art_cc_binary {
         "com.android.art",
         "com.android.art.debug",
         "test_broken_com.android.art",
+        "//apex_available:platform",
     ],
 }
 
