diff --git a/libartservice/service/Android.bp b/libartservice/service/Android.bp
index ad2033a..3c36b77 100644
--- a/libartservice/service/Android.bp
+++ b/libartservice/service/Android.bp
@@ -111,6 +111,7 @@ java_sdk_library {
     // This target is named 'service-art' to conform to the naming conventions
     // for JAR files in the System Server.
     name: "service-art",
+    unsafe_ignore_missing_latest_api: true,
     defaults: [
         "service-art-defaults",
         "framework-system-server-module-optimize-defaults",
@@ -118,6 +119,7 @@ java_sdk_library {
     permitted_packages: ["com.android.server.art"],
     visibility: [
         "//art:__subpackages__",
+        "//frameworks/base/api",
         "//frameworks/base/services/core",
     ],
     apex_available: [
