diff --git a/sdk/ModuleDefaults.bp b/sdk/ModuleDefaults.bp
index d8d95b9..8300df4 100644
--- a/sdk/ModuleDefaults.bp
+++ b/sdk/ModuleDefaults.bp
@@ -77,7 +77,7 @@ java_defaults {
     // but it cannot yet be turned on because some usages have not been cleaned up.
     // TODO(b/156126315) - Remove when no longer needed.
     api_lint: {
-        enabled: true,
+        enabled: false,
     },
 
     // The API scope specific properties.
@@ -208,7 +208,7 @@ stubs_defaults {
             removed_api_file: "api/removed.txt",
         },
         api_lint: {
-            enabled: true,
+            enabled: false,
         },
     },
     dist: {
