--- a/build/config/components/ets_frontend/ets2abc_config.gni
+++ b/build/config/components/ets_frontend/ets2abc_config.gni
@@ -188,8 +188,8 @@ template("generate_static_abc") {
     ]
     external_deps = external_dependencies
     external_deps += [
-      ohos_ets_etsstdlib_deps,
-      ohos_ets_stdlib_deps,
+      # ohos_ets_etsstdlib_deps  # patched,
+      # ohos_ets_stdlib_deps  # patched,
     ]
 
     args = [
@@ -218,1 +218,1 @@ template("generate_static_abc") {
-      deps += [ ohos_ets_libes2panda_public_deps ]
+      # deps += [ ohos_ets_libes2panda_public_deps ]  # patched
@@ -219,8 +219,8 @@ template("generate_static_abc") {
       external_deps += [
-        ohos_ets_api_deps,
-        ohos_ets_arkts_deps,
-        ohos_ets_build_system_deps,
-        ohos_ets_kits_deps,
-        ohos_ets_koala_wrapper_deps,
-        ohos_ets_ui_plugins_deps,
+        # ohos_ets_api_deps  # patched out,
+        # ohos_ets_arkts_deps  # patched out,
+        # ohos_ets_build_system_deps  # patched,
+        # ohos_ets_kits_deps  # patched out,
+        # ohos_ets_koala_wrapper_deps  # patched,
+        # ohos_ets_ui_plugins_deps  # patched out,
       ]
@@ -298,8 +298,8 @@ template("generate_static_abc") {
       }
     } else if (defined(invoker.base_url)) {
       external_deps += [
-        ohos_ets_api_deps,
-        ohos_ets_arkts_deps,
+        # ohos_ets_api_deps  # patched out,
+        # ohos_ets_arkts_deps  # patched out,
       ]
       args += [
         "--base-url",
