--- a/foundation/bundlemanager/bundle_framework/services/bundlemgr/src/installd/installd_operator.cpp
+++ b/foundation/bundlemanager/bundle_framework/services/bundlemgr/src/installd/installd_operator.cpp
@@ -231,7 +231,7 @@
             return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') ||
                 (c >= '0' && c <= '9') || c == '.' || c == '_';
         });
-    if (!validAndroidPackage || !InstalldOperator::IsValidBundleName(bundleName)) {
+    if (!validAndroidPackage || !IsValidBundleName(bundleName)) {
         return false;
     }
     const std::string stageRoot = prefix + bundleName;
