--- a/foundation/bundlemanager/bundle_framework/services/bundlemgr/include/bundle_framework_services_ipc_interface_code.h
+++ b/foundation/bundlemanager/bundle_framework/services/bundlemgr/include/bundle_framework_services_ipc_interface_code.h
@@ -100,6 +100,7 @@
     SET_FILE_CON_FORCE = 73,
     STOP_SET_FILE_CON = 74,
     GET_BUNDLE_FILE_COUNT = 75,
+    COPY_VERIFIED_APK_FROM_FD = 77,
 };
 
 } // namespace AppExecFwk
