--- a/foundation/ability/ability_runtime/services/abilitymgr/src/ability_record.cpp
+++ b/foundation/ability/ability_runtime/services/abilitymgr/src/ability_record.cpp
@@ -2342,6 +2342,7 @@
 {
     std::lock_guard guard(wantLock_);
     missionId_ = missionId;
+    lifeCycleStateInfo_.missionId = missionId_;
     want_.RemoveParam(KEY_MISSION_ID);
     want_.SetParam(KEY_MISSION_ID, missionId_);
 }
