diff --git a/libs/hwui/renderthread/CacheManager.h b/libs/hwui/renderthread/CacheManager.h
index 5e43ac20..d8d6645a 100644
--- a/libs/hwui/renderthread/CacheManager.h
+++ b/libs/hwui/renderthread/CacheManager.h
@@ -17,7 +17,7 @@
 #ifndef CACHEMANAGER_H
 #define CACHEMANAGER_H
 
-#ifdef __ANDROID__ // Layoutlib does not support hardware acceleration
+#if 1 // OH adapter: always enable hardware acceleration
 #include <GrDirectContext.h>
 #endif
 #include <SkSurface.h>
@@ -44,7 +44,7 @@ class CanvasContext;
 
 class CacheManager {
 public:
-#ifdef __ANDROID__ // Layoutlib does not support hardware acceleration
+#if 1 // OH adapter: always enable hardware acceleration
     void configureContext(GrContextOptions* context, const void* identity, ssize_t size);
 #endif
     void trimMemory(TrimLevel mode);
@@ -74,14 +74,14 @@ private:
     void scheduleDestroyContext();
     void cancelDestroyContext();
 
-#ifdef __ANDROID__ // Layoutlib does not support hardware acceleration
+#if 1 // OH adapter: always enable hardware acceleration
     void reset(sk_sp<GrDirectContext> grContext);
 #endif
     void destroy();
 
     RenderThread& mRenderThread;
     const MemoryPolicy& mMemoryPolicy;
-#ifdef __ANDROID__ // Layoutlib does not support hardware acceleration
+#if 1 // OH adapter: always enable hardware acceleration
     sk_sp<GrDirectContext> mGrContext;
 #endif
 
