--- a/foundation/bundlemanager/bundle_framework/services/bundlemgr/src/ipc/extract_param.cpp
+++ b/foundation/bundlemanager/bundle_framework/services/bundlemgr/src/ipc/extract_param.cpp
@@ -29,13 +29,17 @@
 constexpr const char* TYPE_RESOURCE = "Resource";
 constexpr const char* TYPE_RES_FILE = "ResFile";
 constexpr const char* TYPE_HNPS_FILE = "HnpsFile";
+constexpr const char* TYPE_APK_RESOURCES_HAP = "ApkResourcesHap";
+constexpr const char* TYPE_APK_NATIVE_SO = "ApkNativeSo";
 constexpr const char* TYPE_OTHER = "Other";
 const ExtractFileType ARGS_MAP_KEY[] = {
     ExtractFileType::ALL, ExtractFileType::SO, ExtractFileType::AN, ExtractFileType::PATCH, ExtractFileType::AP,
     ExtractFileType::RESOURCE, ExtractFileType::RES_FILE, ExtractFileType::HNPS_FILE,
+    ExtractFileType::APK_RESOURCES_HAP, ExtractFileType::APK_NATIVE_SO,
 };
 constexpr const char* ARGS_MAP_VALUE[] = {
     TYPE_ALL, TYPE_SO, TYPE_AN, TYPE_PATCH, TYPE_AP, TYPE_RESOURCE, TYPE_RES_FILE, TYPE_HNPS_FILE,
+    TYPE_APK_RESOURCES_HAP, TYPE_APK_NATIVE_SO,
 };
 
 std::string GetExtractFileTypeStrVal(const ExtractFileType &extractFileType)
