--- a/foundation/ability/ability_runtime/services/appmgr/include/remote_client_manager.h
+++ b/foundation/ability/ability_runtime/services/appmgr/include/remote_client_manager.h
@@ -63,6 +63,8 @@ public:
 
     std::shared_ptr<AppSpawnClient> GetNativeSpawnClient();
 
+    std::shared_ptr<AppSpawnClient> GetAndroidSpawnClient();
+
     std::shared_ptr<AppSpawnClient> GetHybridSpawnClient();
 
 private:
@@ -71,6 +73,7 @@ private:
     std::shared_ptr<AppSpawnClient> nwebSpawnClient_;
     std::shared_ptr<AppSpawnClient> cjAppSpawnClient_;
     std::shared_ptr<AppSpawnClient> nativeSpawnClient_;
+    std::shared_ptr<AppSpawnClient> androidSpawnClient_;
     std::shared_ptr<AppSpawnClient> hybridSpawnClient_;
 };
 }  // namespace AppExecFwk
