--- a/foundation/ability/ability_runtime/services/appmgr/src/remote_client_manager.cpp
+++ b/foundation/ability/ability_runtime/services/appmgr/src/remote_client_manager.cpp
@@ -26,6 +26,7 @@ RemoteClientManager::RemoteClientManager()
     : appSpawnClient_(std::make_shared<AppSpawnClient>()), nwebSpawnClient_(std::make_shared<AppSpawnClient>(true)),
     cjAppSpawnClient_(std::make_shared<AppSpawnClient>("cjappspawn")),
     nativeSpawnClient_(std::make_shared<AppSpawnClient>("nativespawn")),
+    androidSpawnClient_(std::make_shared<AppSpawnClient>("appspawn-x")),
     hybridSpawnClient_(std::make_shared<AppSpawnClient>("hybridspawn"))
 {}
 
@@ -76,6 +77,14 @@ std::shared_ptr<AppSpawnClient> RemoteClientManager::GetNativeSpawnClient()
     return nativeSpawnClient_;
 }
 
+std::shared_ptr<AppSpawnClient> RemoteClientManager::GetAndroidSpawnClient()
+{
+    if (!androidSpawnClient_) {
+        return nullptr;
+    }
+    return androidSpawnClient_;
+}
+
 std::shared_ptr<AppSpawnClient> RemoteClientManager::GetHybridSpawnClient()
 {
     if (hybridSpawnClient_) {
