--- AssetManager2.cpp.orig
+++ AssetManager2.cpp
@@ -1434,12 +1434,6 @@
 Theme::Theme(AssetManager2* asset_manager) : asset_manager_(asset_manager) {
 }

 Theme::~Theme() = default;
-
-struct Theme::Entry {
-  ApkAssetsCookie cookie;
-  uint32_t type_spec_flags;
-  Res_value value;
-};

 base::expected<std::monostate, NullOrIOError> Theme::ApplyStyle(uint32_t resid, bool force) {
   ATRACE_NAME("Theme::ApplyStyle");
